### PR TITLE
DI using Dagger

### PR DESCRIPTION
**Pull request by saket**

### CODE REVIEW

File: app/src/main/java/org/resolvetosavelives/red/RedApp.kt [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
 class RedApp : Application() {
 
+  companion object {
+    lateinit var appComponent: AppComponent
+  }
+
   override fun onCreate() {
     super.onCreate()
 
     Timber.i("Planing timber tree")
```
Comment (pratul) [inline app/src/main/java/org/resolvetosavelives/red/RedApp.kt]: unnecessary?

File: app/src/main/java/org/resolvetosavelives/red/widgets/Views.kt [hunk 2]
```diff
@@ -8,6 +8,6 @@ fun EditText.showKeyboard() {
   post({
     this.requestFocus()
     val inputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-    inputMethodManager.showSoftInput(this, InputMethodManager.SHOW_FORCED)
+    inputMethodManager.showSoftInput(this, InputMethodManager.SHOW_IMPLICIT)
```
Comment (pratul) [inline app/src/main/java/org/resolvetosavelives/red/widgets/Views.kt]: Why was this changed, exactly?

File: app/src/main/java/org/resolvetosavelives/red/RedApp.kt (hunk 1 above)
Comment (saket) [reply app/src/main/java/org/resolvetosavelives/red/RedApp.kt]: Leftover, yes. Will remove this.

File: app/src/main/java/org/resolvetosavelives/red/widgets/Views.kt (hunk 2 above)
Comment (saket) [reply app/src/main/java/org/resolvetosavelives/red/widgets/Views.kt]: I was initially using `SHOW_FORCED` because the keyboard wasn't showing up for one screen with `SHOW_IMPLICIT`. But I later saw that the keyboard never gets removed when the screen is changed.
